### PR TITLE
Extract ReadBundler to utils

### PR DESCRIPTION
**Pull request by standage**
See #1036. The `clean_up_reads` function was unnecessarily verbose, so I integrated it with the ReadBundler constructor (the only place it was called) and added a couple of property methods.

I added a simple test, which should provide coverage of the entire ReadBundler class sans the `coverages` method.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [ ] <strike>Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.</strike>
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] <strike>Do the changes respect streaming IO? (Are they
  tested for streaming IO?)</strike>
- [x] Is the Copyright year up to date?

### CODE REVIEW

**Comment by standage** [comment]
Ready for review @betatim @camillescott @ctb @luizirber

**Comment by betatim** [comment]
I vote for docs + edge case testing. Low hanging fruit issue for the further refactor.

**Comment by standage** [comment]
Added tests, documentation, and new issue for further refactoring (#1466). Should be ready for merge now.

File: tests/test_read_handling.py [hunk 1]
```diff
@@ -777,3 +777,46 @@ def test_extract_paired_reads_5_stdin_error():
...
+    assert bundle.num_reads == 2
+    assert bundle.agg_length == 200
+    assert bundle.cleaned_reads[0] == cleaned_reads[0]
+    assert bundle.cleaned_reads[1] == cleaned_reads[1]
+
+    for (rd, cln), raw, tstcln in zip(bundle.both(), raw_reads, cleaned_reads):
+        assert rd.sequence == raw
+        assert cln == tstcln
+
+    infile = utils.get_test_data('single-read.fq')
```
Comment (ctb) [inline tests/test_read_handling.py]: In general each test should look at one kind of behavior, so that you don't fix one thing and still have the test failing. So, please break this up into 4 functions.

File: khmer/utils.py [hunk 2]
```diff
@@ -249,5 +249,25 @@ def write_record_pair(read1, read2, fileobj):
...
+
+    def both(self):
+        return zip(self.reads, self.cleaned_reads)
+
+    @property
+    def num_reads(self):
+        return len(self.cleaned_reads)
+
+    @property
+    def agg_length(self):
```
Comment (ctb) [inline khmer/utils.py]: we don't use 'agg' (presumably for 'aggregate') anywhere else in the code base - how about 'sum_length' or 'total_length'?

**Comment by standage** [comment]
Same!
